### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Artem-user-19**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,5 +1,69 @@
...
+
+class BaseRobot:
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: Optional[List[int]] = None) -> None:
+        self.name = name
+        self.weight = weight
+        self.coords = coords if coords is not None else [0, 0]
```
Comment (pavlejviki) [inline app/main.py:17]:
```suggestion
        self.coords = coords if coords else [0, 0]
```

File: app/main.py [hunk 2]
```diff
@@ -1,5 +1,69 @@
...
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: Optional[List[int]] = None) -> None:
+        if coords is None:
+            coords = [0, 0, 0]
```
Comment (pavlejviki) [inline app/main.py:42]:
```suggestion
            coords = coords or [0, 0, 0]
```

**Review by pavlejviki — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1,5 +1,69 @@
...
         self.weight = weight
 
-# write your code here
+
+class BaseRobot:
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: Optional[List[int]] = None) -> None:
```
Comment (Dimosphen1) [inline app/main.py:14]: Place `) -> None:` on a new line to match the code style when there are no arguments on the same line with open and closing parenthesis.

File: app/main.py [hunk 4]
```diff
@@ -1,5 +1,69 @@
...
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: Optional[List[int]] = None) -> None:
```
Comment (Dimosphen1) [inline app/main.py:40]: Place `) -> None:` on a new line to match the code style when there are no arguments on the same line with open and closing parenthesis.

File: app/main.py [hunk 5]
```diff
@@ -1,5 +1,69 @@
...
+
+
+class DeliveryDrone(FlyingRobot):
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            max_load_weight: int,
+            current_load: Optional[Cargo] = None,
+            coords: Optional[List[int]] = None) -> None:
```
Comment (Dimosphen1) [inline app/main.py:59]: Place `) -> None:` on a new line to match the code style when there are no arguments on the same line with open and closing parenthesis.

File: app/main.py [hunk 6]
```diff
@@ -1,5 +1,69 @@
...
+
+class FlyingRobot(BaseRobot):
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: Optional[List[int]] = None) -> None:
+        if coords is None:
+            coords = [0, 0, 0]
+        super().__init__(name, weight, coords)
```
Comment (Dimosphen1) [inline app/main.py:43]: You could define the `coords` directly in the `__init__`, there is no need to allocate a separate variable for that
```suggestion
        super().__init__(name, weight, coords or [0, 0, 0])
```

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested.

File: app/main.py [hunk 7]
```diff
@@ -1,5 +1,69 @@
...
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: Optional[List[int]] = None) -> None:
+        if coords is None:
+            coords = [0, 0, 0]
```
Comment (Dimosphen1) [inline app/main.py:42]: Delete this part, there is no need for it since this logic is implemented on line 42.

File: app/main.py (hunk 4 above)
Comment (Dimosphen1) [reply app/main.py:40]: This comment was not implemented.

File: app/main.py (hunk 5 above)
Comment (Dimosphen1) [reply app/main.py:59]: This comment was not implemented.

File: app/main.py (hunk 3 above)
Comment (Dimosphen1) [reply app/main.py:14]: This comment was not implemented.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Review all comments upon the next review request.

**Pull request closed without merging**